### ISSUE TITLE
Lots of false positives when scanning the gateway-api repo

### ISSUE THREAD

**Issue by dprotaso** [Antonboom/testifylint#67]
```sh
git clone https://github.com/kubernetes-sigs/gateway-api.git
cd gateway-api
git checkout 4f71b2306e7df666a8528dda59ede6620ad0e4e2
go install github.com/Antonboom/testifylint@latest
testifylint ./...
```

You'll see tons of false positives - I've disabled the linter so it's important to checkout the repo at that SHA

```
/gateway-api/conformance/utils/kubernetes/certificate.go:45:2: empty: use require.NotEmpty
/gateway-api/conformance/utils/http/mirror.go:47:4: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-modify-listeners.go:152:4: len: use require.Lenf
/gateway-api/conformance/tests/gateway-static-addresses.go:81:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:82:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:98:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:103:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:119:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:124:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:136:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:137:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:138:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/gateway-static-addresses.go:139:3: go-require: require must only be used in the goroutine running the test function
/gateway-api/conformance/tests/httproute-reference-grant.go:72:3: go-require: require must only be used in the goroutine running the test function
```

There are no go-routines in `/gateway-api/conformance/tests/gateway-static-addresses.go`

See:  https://github.com/kubernetes-sigs/gateway-api/blob/4f71b2306e7df666a8528dda59ede6620ad0e4e2/conformance/tests/gateway-static-addresses.go

**Comment by Antonboom** [comment]
hi, @dprotaso!

thank you for request.

not false positive:
```go
// kubernetes/certificate.go:45:2: empty: use require.NotEmpty
require.Greater(t, len(hosts), 0, "require a non-empty hosts for Subject Alternate Name values")
```

not false positive:
```go
// gateway-modify-listeners.go:152:4: len: use require.Lenf
require.Equalf(t, 2, len(mutate.Spec.Listeners), "the gateway must have 2 listeners")
```

About `go-require` – I will investigate.
Anyway you can just disable one checker, not overall linter 👌 

https://golangci-lint.run/usage/linters/#testifylint

**Comment by Antonboom** [comment]
@dprotaso, thanks for this tricky case

fixed!

```
/private/tmp/gateway-api 4f71b230
$ testifylint ./...        
/private/tmp/gateway-api/conformance/utils/kubernetes/certificate.go:45:2: empty: use require.NotEmpty
/private/tmp/gateway-api/conformance/utils/http/mirror.go:47:4: go-require: require must only be used in the goroutine running the test function
/private/tmp/gateway-api/conformance/tests/gateway-modify-listeners.go:152:4: len: use require.Lenf
```

**Comment by dprotaso** [comment]
awesome thanks @Antonboom - yeah some of them are not false positives.

What's the best way to pick up these changes? The project is using golangci-lint

**Comment by Antonboom** [comment]
> What's the best way to pick up these changes? 

golangci-lint will pull new version automatically and these changes will be available in the new golangci-lint's release
